projects/packages/videopress/src/client/block-editor/blocks/video/components/videopress-player/index.native.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * External dependencies
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { SandBox } from '@wordpress/components';
+import { useCallback, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { View, Text } from 'react-native';
+
+/**
+ * VideoPlayer react component
+ *
+ * @param {object} props                 - Component props.
+ * @param {object} props.html            - Player html to render in the sandbox.
+ * @param {object} props.attributes      - Block attributes.
+ * @param {Array} props.scripts          - Scripts to pass trough to the sandbox.
+ * @param {object} props.preview         - oEmbed preview data.
+ * @param {boolean} props.isRequestingEmbedPreview - oEmbed preview data.
+ * @returns {object}                     - React component.
+ */
+export default function VideoPressPlayer( {
+	html,
+	attributes,
+	scripts = [],
+	preview,
+	isRequestingEmbedPreview,
+} ) {
+	const [ videoPlayerHeight, setVideoPlayerHeight ] = useState( 250 );
+
+	// Use preview aspect ratio if available. Otherwise, use the ratio from the attributes.
+	const { height = 0, width = 1 } = preview;
+	const videoRatio = ( height / width ) * 100 || attributes.videoRatio;
+
+	// If we don't have `html` then the player is loading.
+	const loadingVideoPlayer = ! html;
+
+	const embedPreviewRequested = ! isRequestingEmbedPreview;
+
+	// Scale video player height based on the video ratio and layout width.
+	const scaleVideoPlayer = useCallback(
+		( { layout } ) => !! videoRatio && setVideoPlayerHeight( ( layout.width * videoRatio ) / 100 ),
+		[ videoRatio ]
+	);
+
+	return (
+		<View
+			style={ { height: videoPlayerHeight } }
+			onLayout={ ( { nativeEvent } ) => scaleVideoPlayer( nativeEvent ) }
```

Comment:
I recall that the height was automatically set in the `Sandbox` component ([reference](https://github.com/WordPress/gutenberg/blob/b4e9e0aed4910eb7b6f57b67f866b37ea05e78ee/packages/components/src/sandbox/index.native.js#L264-L281)) communicating with the embed content, not sure if we could apply a similar approach for the VideoPress player 🤔 .

Reply:
I _believe_ I was running into issues with the aspect ratio of the videos, not just the height. I'll play around with this to see if I can get the Sandbox to handle it correctly

Reply:
Ok I figured out what's going on here. I was following the web version a bit too closely I think. There is a "loading" view that needs some height.  I think the web version also needs to handle responsive UI changes so I'm assuming the height needs to be handled explicitly.

I can handle the "is loading" state better which simplifies a lot of things. 
Thanks for pointing that out!